homeassistant/components/ecoforest/__init__.py

```diff
@@ -0,0 +1,58 @@
+"""The Ecoforest integration."""
+from __future__ import annotations
+
+import logging
+
+import async_timeout
+import httpx
+from pyecoforest.api import EcoforestApi
+from pyecoforest.exceptions import EcoforestAuthenticationRequired
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN
+from .coordinator import EcoforestCoordinator
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Ecoforest from a config entry."""
+
+    host = entry.data[CONF_HOST]
+    auth = httpx.BasicAuth(entry.data[CONF_USERNAME], entry.data[CONF_PASSWORD])
+    api = EcoforestApi(host, auth)
+
+    try:
+        async with async_timeout.timeout(10):
```

Comment:
Also, not a showstopper, but I think it would be nice if the timeout mechanism could move to the library at some point as it's a part of how to communicate with the device.